apps/govquests-frontend/src/app/[id]/page.tsx

```diff
@@ -0,0 +1,85 @@
+"use client";
+
+import type { Quest } from "@/types/quest";
+import { getQuestById } from "@/utils/api";
+import { ExternalLinkIcon } from "@radix-ui/react-icons";
+import Link from "next/link";
+import type React from "react";
+import { useEffect, useState } from "react";
+
+const QuestDetails: React.FC = () => {
+  const [quest, setQuest] = useState<Quest | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchQuest = async () => {
```

Comment:
why don't you use a server-side query to fetch the quest id that is in the URL? e.g in the op-ai topic-details page we fetch on the server side and then render a client side component

https://github.com/bleu/op-ai-tools/blob/main/www/src/app/forum/topic/%5BtopicId%5D/page.tsx